src/Init/Data/Nat/Gcd.lean

```diff
@@ -10,6 +10,15 @@ import Init.RCases
 
 namespace Nat
 
+/--
+Computes the greatest common divisor of two natural numbers.
+
+This reference implementation via the Euclidean algorithm
+is overridden in both the kernel and the compiler to efficiently
+evaluate using the "bignum" representation (see `Nat`).
+The definition provided here is the logical model
+(and it is soundness-critical that they coincide).
+-/
```

Comment:
Code examples would be useful, incl. edge cases.

Reply:
Added, thanks.